graph-partitioning/README.rst

```diff
@@ -0,0 +1,82 @@
+Graph Partitioning Demo
+=======================
+In this demo, we explore the graph partitioning problem.  This is an interesting problem from graph theory with a wide variety of applications (see e.g. [3]_).
+
+For example, suppose that we have a group of people that we need to split into two equal-sized teams.  For each pair of people there is a relationship:  friends or enemies.  We want to have as few friends on opposite teams as possible.  How can we solve this problem?
+
+One way to solve this problem is with the graph partitioning problem.  If we think of our group of people as a graph (a node/vertex for each person), and draw an edge between people that are friends, we have a model of our network.  If we look for a minimum cut in our graph, then we are looking for a way to split the nodes into two groups so that there are as few edges as possible between the groups.  In our group of peoople, this means that we have two teams with as few friends as possible on different teams.
+
+We will run the graph partitioning problem on an Erdos-Renyi random graph with probability 0.2 on 40 nodes to find the best way of splitting the graph into two subsets to minimize the number of cut edges.
+
+Usage
+-----
+To run the demo, type:
+::
+  python graph_partitioning.py
+
+After running, output will be printed to the command line that states whether or not the best (lowest energy) solution found is a valid partition of two equal-sized sets.  If the solution is valid (feasible), then the number of cut edges for that solution is displayed.
+
+Code Overview
+-------------
+The code implements a QUBO formulation of this problem.
+
+The answer that we are looking for is a partition of the nodes in the graph, so we will assign a binary variable for each node, i.e. variable x_i denotes whether node i is in one subset (call it Subset 0) or the other (Subset 1).
+
+The objective function that we are looking to optimize is minimizing the number of cut edges.  To count how many cut edges we have given a partition of the nodes (assignment of our binary variables), we consider a single edge in a graph in the table below.  We only want to count an edge if the endpoints are in different subsets, and so we assign a 1 for the edge column in this case and a 0 otherwise.
+
```

Comment:
Could this be rephrased so it's clear from the start that we're explaining how the calculation is done by showing an example in the table? Something like, "The table below shows how we count the cut edges for a given graph partition (assignment of values to our binary variables). Columns :math:`x_i`: and :math:x_j` are two nodes; column edge (i, j) represents ..."

Reply:
Added